File: server/Entities/Hero.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Attribute;

namespace Entities
{
    public class Hero
    {
        [Required]
        public Guid Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        [Ability]
        public string Ability { get; set; }
        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public string SuitColor { get; set; }
        [Required]
        public double StartingPower { get; set; }
        public double CuretPower { get; set; }
        public AppUser Trainer { get; set; }
        [Required]

        public String TrainerId { get; set; }
#nullable enable

        public ICollection<Training>? TrainHistory { get; set; }


    }
}

File: client/src/app/model/hero.ts
export interface Hero {
  id: string;
  name: string;
  firstTraining: Date;
  suitColor: string;
  startingPower: number;
  curetPower: number;
  trainedTodayTimes: number;
}
export interface HeroCreate {
  name: string;
  suitColor: string;
  power: number;
}
